testutils/file_utils.go

```diff
@@ -37,7 +40,77 @@ func BuildTempModule(tb testing.TB, dir string) string {
 	if tb.Failed() {
 		tb.Fatalf("failed to build temporary module for testing")
 	}
-	return modPath
+	return exePath
+}
+
+// BuildTempModuleWithFirstRun will attempt to build the module in the provided directory and put the
+// resulting executable binary into a temporary directory. After building, it will also copy "meta.json"
+// and "first_run.sh" into the same temporary directory. It is assumed that these files are in the
+// provided module directory. If successful, this function will return the path to the executable binary.
+func BuildTempModuleWithFirstRun(tb testing.TB, modDir string) string {
+	// tb.Helper()
+
+	exePath := BuildTempModule(tb, modDir)
+	exeDir := filepath.Dir(exePath)
+
+	type copyOp struct {
+		src string
+		dst string
+	}
+
+	for _, cp := range []copyOp{
+		// TODO(RSDK-9151): Having a `meta.json` in the testmodule directory results in
+		// unintended behavior changes across our test suite. To avoid this, we name the
+		// file `first_run_meta.json` in the source test module directory and rename it in the
+		// destination directory that contains the built executable. This is fine for now
+		// but we should investigate and understand why this happens.
```

Comment:
do we know why now? is this related to files getting deleted because of the tests?

Reply:
i still don't know why this happened - think this is a different issue than the issue you are referring to. i can try to revert this be `meta.json` to see if we still encounter failures, but I suspect we will. in any case, i think it's helpful to have a "labeled" meta.json that somewhat ties it to a particular test.

another potential approach here is to have the helper function generate a meta.json for testing instead of copying from a file in scm.

Reply:
ya, it's ok for now